src/main/java/org/carlspring/cloud/storage/s3fs/S3SeekableByteChannel.java

```diff
@@ -71,7 +71,7 @@ else if (!exists && !this.options.contains(StandardOpenOption.CREATE_NEW) && !th
 
         if(tempFileRequired)
         {
-            tempFile = Files.createTempFile("temp-s3-", key.replaceAll("/", "_"));
+            tempFile = Files.createTempFile("temp-s3-", fileName == null ? key : fileName.toString());
```

Comment:
Honestly, I would just leave this as:
```suggestion
            tempFile = Files.createTempFile("s3fs-");
```
simply because if the `fileName` is again, for some reason, larger than what UNIX can handle it would fail with the same exception, would it not?

Have you tried listing a "directory" which holds a file with very long name? Would this also throw an exception?

Reply:
You are absolutely right, the case that you are saying would also throw an exception.
I've just fixed it. 
BTW, `createTempFile` method also needs a suffix parameter, that's why I've added a `.tmp` extension.

Reply:
Awesome! What about the listing long file names - would that also throw an exception we should address in a follow-up?

Reply:
We should probably check this as well, while we're at it (already).

Reply:
I'm not sure if I understand you... I've created a directory in my S3 bucket; there I've uploaded a file whose name has more than 255 characters, then I've listed the objects in the directory using the AWS SDK v2 API, and I can obtain it without any problem.

Reply:
Then, in this case, you seem to have covered this. Thanks! :)